### ISSUE TITLE
🚀 Feature: Implement Redis Caching for Job Listings to Improve Load Times

### ISSUE THREAD

**Issue by Paribesh01** [code100x/job-board#559]
### 🔖 Feature description

The caching feature aims to improve the performance and responsiveness of the job board application by temporarily storing frequently accessed job listings, search results, and filter queries in a caching layer (using Redis). 

### 🎤 Why is this feature needed ?

Implementing a caching feature in the job board application is crucial for enhancing performance and improving user experience by reducing load times for frequently accessed data.

### ✌️ How do you aim to achieve this?

we will start by setting up a Redis server for storing cached data. We'll adjust how we fetch job listings so that the application first checks the cache for data before going to the database. This will involve creating unique keys for the cached data based on different job filters. We’ll also implement rules for how long data stays in the cache and when it should be removed to ensure users always see up-to-date information.

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/code100x/job-board/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by Paribesh01** [comment]
@VineeTagarwaL-code should we implement this??

**Comment by VineeTagarwaL-code** [comment]
yeah sounds like a good thing to work on, will be easy to implement and stuff just have to think about revalidation and stuff when new job gets added will ask bhayiya about this and get back to you

**Comment by Paribesh01** [comment]
@VineeTagarwaL-code what's the update should I start working??

**Comment by VineeTagarwaL-code** [comment]
good thinking indeed, but i want something else 

we are not using tanstack query for job fetching and stuff, can you migrate our current fetching to tanstack query, it has inbuilt caching and global state solution 

do it only for jobs now, no need to stress about the whole codebase

**Comment by Paribesh01** [comment]
@VineeTagarwaL-code ok, i will start working..

**Comment by Kannav02** [comment]
Hey @Paribesh01 , i can help on this as well if you want me to 

Thank you!
